flink-core/src/main/java/org/apache/flink/util/function/TriConsumerWithException.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.util.function;
+
+import org.apache.flink.annotation.PublicEvolving;
+
+/**
+ * Operation which is performed on three given arguments.
+ *
+ * @param <S> type of the first argument
+ * @param <T> type of the second argument
+ * @param <U> type of the third argument
+ */
+@PublicEvolving
+@FunctionalInterface
+public interface TriConsumerWithException<S, T, U, E extends Throwable> {
```

Comment:
How about more informative names for type parameters here?
For example, `<First, Second, Third, Err extends Throwable>`.

Reply:
Current names are inline with other such classes `TriFunctionWithException`, `ThrowingConsumer` and the java's interfaces: `Consumer`, `BiFunction` etc.

Reply:
When implementing **this** interface, I'd look at **this** interface first, and maybe then at the related classes.
However, consistent naming is also a valid point.